src/main/java/fi/helsinki/opintoni/localization/TeachingLanguages.java

```diff
@@ -0,0 +1,155 @@
+        Language.EN.getCode(), "German")),
+    DA(Code.DA, ImmutableMap.of(
+        Language.FI.getCode(), "tanska",
+        Language.SV.getCode(), "danska",
+        Language.EN.getCode(), "Danish")),
+    RU(Code.RU, ImmutableMap.of(
+        Language.FI.getCode(), "venäjä",
+        Language.SV.getCode(), "ryska",
+        Language.EN.getCode(), "Russian")),
+    ET(Code.ET, ImmutableMap.of(
+        Language.FI.getCode(), "viro",
+        Language.SV.getCode(), "estniska",
+        Language.EN.getCode(), "Estonian")),
+    FI_1ST_SC(Code.FI_1ST_SC, ImmutableMap.of(
+        Language.FI.getCode(), "äidinkieli (suomi), puheviestintä- ja vuorovaikutustaidot",
+        Language.SV.getCode(), "modersmål (finska), retorik",
+        Language.EN.getCode(), "Finnish as a native language, speech communication")),
+    FI_1ST_AW(Code.FI_1ST_AW, ImmutableMap.of(
+        Language.FI.getCode(), "äidinkieli (suomi), akateemiset tekstitaidot",
+        Language.SV.getCode(), "modersmål (finska), akademiskt skrivande",
+        Language.EN.getCode(), "Finnish as a native language, academic writing")),
+    FI_2ND(Code.FI_2ND, ImmutableMap.of(
+        Language.FI.getCode(), "suomi toisena kotimaisena kielenä (finska)",
+        Language.SV.getCode(), "finska som andra inhemska språk",
+        Language.EN.getCode(), "Finnish for native speakers of Swedish")),
+    SV_1ST(Code.SV_1ST, ImmutableMap.of(
+        Language.FI.getCode(), "äidinkieli (ruotsi)",
+        Language.SV.getCode(), "modersmål (svenska)",
+        Language.EN.getCode(), "Swedish as a native language")),
+    SV_2ND(Code.SV_2ND, ImmutableMap.of(
+        Language.FI.getCode(), "ruotsi toisena kotimaisena kielenä",
+        Language.SV.getCode(), "svenska som andra inhemska språk",
+        Language.EN.getCode(), "Swedish for native speakers of Finnish")),
+    SLU(Code.SLU, ImmutableMap.of(
+        Language.FI.getCode(), "pedagoginen yliopistonlehtori",
+        Language.SV.getCode(), "pedagogisk universitetslektor",
+        Language.EN.getCode(), "senior lecturer in university pedagogy"));
+
+    public enum Code {
+        AR("ar"), EN("en"), ES("es"), IT("it"), JA("ja"), ZG("zg"), KO("ko"), PT("pt"), FR("fr"), DE("de"), DA("da"),
+        RU("ru"), ET("et"), FI_1ST_SC("fi_1st_sc"), FI_1ST_AW("fi_1st_aw"), FI_2ND("fi_2nd"), SV_1ST("sv_1st"),
+        SV_2ND("sv_2nd"), SLU("slu");
+
+        private final String code;
+
+        Code(String code) {
+            this.code = code;
+        }
+
+        public static Code fromString(String code) {
+            return Optional.of(Code.valueOf(code.toUpperCase()))
+                .orElseThrow(() -> new IllegalArgumentException(""));
+        }
+
+        public String getCode() {
+            return this.code;
+        }
+    }
+
+    private final Code code;
+    private final ImmutableMap<String, String> localizedNames;
+
+    TeachingLanguages(Code code, ImmutableMap<String, String> localizedNames) {
```

Comment:
Huomasin vasta nyt että tämä enum on nimetty monikossa eikä yksikössä, vaikka enumin nimen olisi hyvä olla aina yksikössä kuten tavallisenkin classin. Koska yksiköllinen nimi menisi päällekkäin samannimisen JPA-entiteetin kanssa, voisi tehdä erillisen `domain.enum`-alipaketin jonne enumit siirtää, jolloin nimikonflikteja ei tule.